tests/phpunit/Persistence/Search/Query/RangeFacetQueryBuilderTest.php

```diff
@@ -0,0 +1,132 @@
+ */
+class RangeFacetQueryBuilderTest extends TestCase {
+
+	private const QUANTITY_PROPERTY = 'P100';
+	private const STRING_PROPERTY = 'P200';
+	private const TIME_PROPERTY = 'P300';
+	private const ITEM_PROPERTY = 'P400';
+
+	public function testBuildsQueryForQuantityRange(): void {
+		$query = $this->newFacetQueryBuilder()->buildQuery(
+			$this->newRangeFacetConfig( self::QUANTITY_PROPERTY ),
+			new PropertyConstraints(
+				new NumericPropertyId( self::QUANTITY_PROPERTY ),
+				min: 42,
+				max: 9001
+			)
+		);
+
+		$this->assertIsRangeQueryWithParams(
+			self::QUANTITY_PROPERTY,
+			[ 'gte' => 42.0, 'lte' => 9001.0 ],
+			$query
+		);
+	}
+
+	private function newFacetQueryBuilder(): RangeFacetQueryBuilder {
+		return new RangeFacetQueryBuilder(
+			$this->newDataTypeLookup()
+		);
+	}
+
+	private function newDataTypeLookup(): InMemoryDataTypeLookup {
+		$dataTypeLookup = new InMemoryDataTypeLookup();
+
+		$types = [
+			self::QUANTITY_PROPERTY => 'quantity',
+			self::STRING_PROPERTY => 'string',
+			self::TIME_PROPERTY => 'time',
+			self::ITEM_PROPERTY => 'wikibase-item'
+		];
+
+		foreach ( $types as $pId => $type ) {
+			$dataTypeLookup->setDataTypeForProperty(
+				new NumericPropertyId( $pId ),
+				$type
+			);
+		}
+
+		return $dataTypeLookup;
+	}
+
+	private function newRangeFacetConfig( string $propertyId ): FacetConfig {
+		return new FacetConfig(
+			new ItemId( 'Q404' ),
+			new NumericPropertyId( $propertyId ),
+			FacetType::RANGE
+		);
+	}
+
+	private function assertIsRangeQueryWithParams( string $propertyId, array $values, Query\AbstractQuery $query ): void {
+		$this->assertInstanceOf( Query\Range::class, $query );
+		$this->assertSame(
+			[
+				'wbfs_' . $propertyId => $values
+			],
+			$query->getParams()
+		);
+	}
+
+	public function testBuildsQueryForDateRange(): void {
+		$query = $this->newFacetQueryBuilder()->buildQuery(
+			$this->newRangeFacetConfig( self::TIME_PROPERTY ),
+			new PropertyConstraints(
+				new NumericPropertyId( self::TIME_PROPERTY ),
+				min: 2000,
+				max: 2010
+			)
+		);
+
+		$this->assertEquals(
+			new Query\Range(
+				'wbfs_' . self::TIME_PROPERTY,
+				[ 'gte' => '2000-01-01', 'lte' => '2010-12-31' ]
+			),
+			$query
+		);
+	}
+
+	public function testDoesNotBuildQueryForStringRange(): void {
+		$query = $this->newFacetQueryBuilder()->buildQuery(
+			$this->newRangeFacetConfig( self::STRING_PROPERTY ),
+			new PropertyConstraints(
+				new NumericPropertyId( self::STRING_PROPERTY ),
+				orSelectedValues: [ '2000', '2010' ]
+			)
+		);
+
+		$this->assertNull( $query );
+	}
+
+	public function testDoesNotBuildQueryForItemRange(): void {
+		$query = $this->newFacetQueryBuilder()->buildQuery(
+			$this->newRangeFacetConfig( self::ITEM_PROPERTY ),
+			new PropertyConstraints(
+				new NumericPropertyId( self::ITEM_PROPERTY ),
+				orSelectedValues: [ 'Q100', 'Q200' ]
+			)
+		);
+
+		$this->assertNull( $query );
+	}
```

Comment:
There does not seem to be any test for the common-ish use case of open ranges (only min or only max)